src/libasr/codegen/asr_to_fortran.cpp

```diff
@@ -1312,29 +1336,57 @@ class ASRToFortranVisitor : public ASR::BaseVisitor<ASRToFortranVisitor>
         this->visit_expr(*x.m_arg);
     }
 
-    // void visit_ComplexRe(const ASR::ComplexRe_t &x) {}
+    void visit_ComplexRe(const ASR::ComplexRe_t &x) {
+        visit_expr(*x.m_arg);
+        s = "real(" + s + ")";
+    }
 
-    // void visit_ComplexIm(const ASR::ComplexIm_t &x) {}
+    void visit_ComplexIm(const ASR::ComplexIm_t &x) {
+        visit_expr(*x.m_arg);
+        s = "aimag(" + s + ")";
+    }
 
     // void visit_CLoc(const ASR::CLoc_t &x) {}
 
     // void visit_PointerToCPtr(const ASR::PointerToCPtr_t &x) {}
 
     // void visit_GetPointer(const ASR::GetPointer_t &x) {}
 
-    // void visit_IntegerBitLen(const ASR::IntegerBitLen_t &x) {}
+    void visit_IntegerBitLen(const ASR::IntegerBitLen_t &x) {
+        visit_expr(*x.m_a);
+        int arg_kind = ASRUtils::extract_kind_from_ttype_t(x.m_type);
+        switch (arg_kind) {
+            case 1: s = "_lpython_bit_length1(" + s + ")"; break;
+            case 2: s = "_lpython_bit_length2(" + s + ")"; break;
+            case 4: s = "_lpython_bit_length4(" + s + ")"; break;
+            case 8: s = "_lpython_bit_length8(" + s + ")"; break;
+            default: {
+                throw LCompilersException("Unsupported Integer Kind: "
+                    + std::to_string(arg_kind));
+            }
+        }
```

Comment:
I think this can be simplified as:
```suggestion
        s = "bit_size(" + s + ")";
```